rest_framework/schemas/openapi.py

```diff
@@ -209,6 +211,33 @@ def _get_pagination_parameters(self, path, method):
 
         return paginator.get_schema_operation_parameters(view)
 
+    def _map_choicefield(self, field):
+        choices = list(OrderedDict.fromkeys(field.choices))  # preserve order and remove duplicates
+        if all(isinstance(choice, bool) for choice in choices):
+            type = 'boolean'
+        elif all(isinstance(choice, int) for choice in choices):
+            type = 'integer'
+        elif all(isinstance(choice, (int, float, Decimal)) for choice in choices):  # `number` includes `integer`
+            # Ref: https://tools.ietf.org/html/draft-wright-json-schema-validation-00#section-5.21
+            type = 'number'
+        elif all(isinstance(choice, str) for choice in choices):
+            type = 'string'
+        else:
+            type = None
+
+        mapping = {
+            # The value of `enum` keyword MUST be an array and SHOULD be unique.
+            # Ref: https://tools.ietf.org/html/draft-wright-json-schema-validation-00#section-5.20
+            'enum': choices
+        }
+
+        # If We figured out `type` then and only then we should set it. It must be string or an array.
```

Comment:
https://swagger.io/docs/specification/data-models/data-types/#mixed-type

This actually must just be a string. JSON Schema allows it to be an array, but OpenAPI does not.

Reply:
Done.